### ISSUE TITLE
fix workaround which does not use target native features on Windows

### ISSUE THREAD

**Issue by andrewrk** [ziglang/zig#508]
~See #302. I had to add a workaround to treat every build on windows as if we were cross compiling for the native target. Once the llvm bug is fixed we can remove this workaround.~

See below.

mailing list message: http://lists.llvm.org/pipermail/llvm-dev/2017-September/117860.html

**Comment by andrewrk** [comment]
Accident, incorrectly tagged this issue in the above commit.

**Comment by suirad** [comment]
Is this not an upstream issue, since it depends on llvm changes?

**Comment by andrewrk** [comment]
I don't think it's an upstream bug anymore. I think it's a problem with Zig's compiler-rt ABI compliance on Windows. Not aligning enough or something like that.

**Comment by emekoi** [comment]
can we add back native features thanks to #2079?

**Comment by andrewrk** [comment]
> can we add back native features thanks to #2079?

I'll run the tests and report back.

**Comment by andrewrk** [comment]
Here's what I get:

```
394/686 behavior.math.test "behavior-windows-x86_64-Debug-bare-multi f128"...test failure
C:\msys64\home\andy\dev\zig\build-release-llvm8\lib\zig\std\testing.zig:149:14: 0x7ff6578a1029 in std.testing.expect (test.obj)
    if (!ok) @panic("test failure");
             ^
C:\msys64\home\andy\dev\zig\test\stage1\behavior\math.zig:538:11: 0x7ff6578b9e88 in behavior.math.test_f128 (test.obj)
    expect(make_f128(1.0) != 1.1);
          ^
C:\msys64\home\andy\dev\zig\test\stage1\behavior\math.zig:527:14: 0x7ff6578a926f in behavior.math.test "behavior-windows-x86_64-Debug-bare-multi f128" (test.obj)
    test_f128();
             ^
C:\msys64\home\andy\dev\zig\build-release-llvm8\lib\zig\std\special\test_runner.zig:13:25: 0x7ff657908ecb in std.special.main (test.obj)
        if (test_fn.func()) |_| {
                        ^
C:\msys64\home\andy\dev\zig\build-release-llvm8\lib\zig\std\special\start.zig:67:49: 0x7ff657908c8f in ??? (test.obj)
    std.os.windows.kernel32.ExitProcess(callMain());
                                                ^
???:?:?: 0x7ffcea2b4034 in ??? (???)


???:?:?: 0x7ffcebfb3691 in ??? (???)



Tests failed. Use the following command to reproduce the failure:
C:\msys64\home\andy\dev\zig\zig-cache\o\M8vBp-VgTY3kAMkREyfL6R-Yssr4zihm0oJrtl40AsiMjH4brNirFNsCaJI9Bstb\test.exe
The following command exited with error code 2147483651:
c:\msys64\home\andy\dev\zig\build-release-llvm8\bin\zig.exe test C:\msys64\home\andy\dev\zig\test\stage1\behavior.zig --test-name-prefix behavior-windows-x86_64-Debug-bare-multi  --cache-dir C:\msys64\home\andy\dev\zig\zig-cache --name test
```

So even the behavior tests do not pass on Windows with native CPU features turned on.

**Comment by LemonBoy** [comment]
Here's a minimal LLVM IR reproducer:

```llvm
; Function Attrs: cold nobuiltin noreturn nounwind
define internal void @bar(i1) unnamed_addr #1 {
    unreachable
}

; Function Attrs: nobuiltin nounwind uwtable
define internal fastcc void @test_f128() unnamed_addr #0 {
Entry:
  %x = alloca fp128, align 16
  store fp128 0xL00000000000000003FFF000000000000, fp128* %x, align 16
  %0 = load fp128, fp128* %x, align 16
  %1 = fcmp une fp128 %0, 0xLEB851EB851EB851F400091EB851EB851
  call fastcc void @bar(i1 %1)
  ret void
}
```

Compile it with the following command:

```
llc -O0 -mcpu=ivybridge -mtriple x86_64-windows-msvc -filetype asm -o /tmp/foo.s /tmp/foo.ll
```

You can easily notice how the generated assembly has only a single `fp128` constant even though there are two of them in the code above, and the two are definitely not the same value.

The boring news is that the problem is already fixed on LLVM 10.

**Comment by andrewrk** [comment]
> The boring news is that the problem is already fixed on LLVM 10.

:tada: 

Thanks for checking this out. Glad to hear we can do nothing and have this solved in ~6 months.

**Comment by andrewrk** [comment]
If it's obvious what commit fixed it in trunk, we can ask for it to be a 9.0.1 release blocker.
